README.md

```diff
@@ -98,8 +98,7 @@ If the above doesn't fix the problem, open your `.bash_profile` and add the foll
 
 ```
 - name: nvm
-  shell: >
-    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.0/install.sh | bash
```

Comment:
can you elaborate how this is different?

Reply:
Sorry I haven't provided more context.

If you use [ansible-lint](https://docs.ansible.com/ansible-lint/), it'll fail with an error because of several linting rules violated on this task (list of all rules [here](https://docs.ansible.com/ansible-lint/rules/default_rules.html)):
- E303 `curl` used in place of [`get_url`](https://docs.ansible.com/ansible/latest/modules/get_url_module.html) or [`uri`](https://docs.ansible.com/ansible/latest/modules/uri_module.html) module
-- the linter detects you are executing a specific command to download a file and offers a solution _native_ for Ansible using a built-in module
- E306 Shells that use pipes should set the pipefail option
-- Shell only looks at the exit code of the last command you execute. That means without `pipefail` such liner may exit with code 0, while it actually failed, but it gets lost in the pipes.

The alternative I suggest does not suffer from these errors. It does not make the linter think a file is being download (technically it is, but the aim is to execute it). And there are no pipes, so it `curl` fails, it all fails.

Reply:
I vastly prefer the direct curl call to the indirect one, and I don't think ansible-lint should prompt any changes outside an ansible task (but I also prefer consistency with the other commands, so I think you made the right call by changing them all)

The pipefail point is interesting, but doesn't seem like a big enough problem to warrant the inelegance of the bash -c approach :-/

Reply:
Another way to have cleaner Ansible would be to have two tasks. One using `get_url` to fetch the install script e.g. to `/tmp` and second `command` to execute it. Optionally a third one can remove the file from `/tmp` again.

Reply:
in ansible, can the pipefail option be set?

Reply:
Sure. What the `shell` content is run in a shell. So you can change it to:
```yaml
shell: |
  set -o pipefail
  curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.0/install.sh | bash
```

To address the other commands also, they could be changed to use sub-shell, probably something like this
```shell
(set -o pipefail && curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.0/install.sh | bash)
```

Actually that sub-shell could be what you place into Ansible, so it would look like this:
```yaml
shell: (set -o pipefail && curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.0/install.sh | bash)
```